src/service-object.ts

```diff
@@ -178,8 +171,6 @@ class ServiceObject<T = any> extends EventEmitter {
     this.methods = config.methods || {};
     this.interceptors = [];
     this.Promise = this.parent ? this.parent.Promise : undefined;
-    this.requestModule =
```

Comment:
It may be nice to throw if `config.requestModule` is set.

Reply:
I seem to remember @stephenplusplus smacking my hands last time I did something like that in this exact spot :)

Reply:
Haha. Well, throwing developer errors hasn't been something we have done before. We only throw errors that make sense to the user, and leave it to our various testing mechanisms to cover how we use our own libraries.

Reply:
This is a different use-case though. We are detecting that the user didn't read the changelog and is passing a config flag that we are going to silently ignore. It would be nice to tell them that there was a breaking change. I don't have objections to the general disposition about throwing errors in these libraries; but this is different.

Reply:
The users are us, though.

Reply:
I am wildly irresponsible and really rely on TypeScript types and aggressive exceptions to save me 🤷‍♂️

Reply:
I've seen (and like) the pattern of asserting that arguments/config are correct at the start of a function, so that a client fails as early as possible when interacting with the library (vs., having undefined behavior that bubbles as an error like can't access property of undefined).

Having said this, given we're calling this a breaking change and, like @stephenplusplus says, this is an interface probably mainly used by ourselves, I'd argue against the throw (which we'll then wonder about why the heck it's in the codebase for the next 5 years).